contrib/rootless-cni-infra/Containerfile

```diff
@@ -33,3 +33,5 @@ COPY --from=dnsname /dnsname /opt/cni/bin
 COPY rootless-cni-infra /usr/local/bin
 ENV CNI_PATH=/opt/cni/bin
 CMD ["sleep", "infinity"]
+
+ENV CNI_ROOTLESS_INFRA_VERSION=1
```

Comment:
`ROOTLESS_CNI_INFRA_VERSION`

(btw, this env isn't propagated to the container due to a bug😭 : https://github.com/containers/podman/blob/fd7cdb25027bb33c33eacb99f1a02838eca5d684/libpod/rootless_cni_linux.go#L261-L263)

Reply:
Works on my end:
```
libpod (fix-7617) $ podman run --rm quay.io/libpod/rootless-cni-infra:1-amd64 printenv
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
TERM=xterm
container=podman
CNI_PATH=/opt/cni/bin
CNI_ROOTLESS_INFRA_VERSION=1
HOSTNAME=6c0e091712df
HOME=/root

libpod (fix-7617) $ podman run --rm quay.io/libpod/rootless-cni-infra:1-amd64 /usr/local/bin/rootless-cni-infra version
{"version": "1"}
```

Reply:
When podman creates the container as the infra, the ENV is not propagated from Containerfile.
(`podman create foo`, `podman run --network foo something`, `podman exec rootless-cni-infra rootless-cni-infra version`)

Reply:
That sounds like a bug we should fix. Did you open an issue for  it?

Reply:
I think I can fix that up in this PR

Reply:
I added a third commit to extract the env and cmd from the image (if specified).